### PR TITLE
[SW-2143] bin/build-kubernetes-images.sh should get spark version from $SPARK_HOME directory

### PR DESCRIPTION
**Pull request by mn-mikke**
Since the current logic gets the version from gradle.properties. The Spark version doesn’t have to match the current environment.

### CODE REVIEW

File: bin/build-kubernetes-images.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@ fi
 TOPDIR=$(cd "$(dirname "$0")/.." || exit; pwd)
 
 source "$TOPDIR/bin/sparkling-env.sh"
+export SPARK_VERSION=$(grep Spark  "$SPARK_HOME/RELEASE" | cut -d' ' -f2)
```
Comment (jakubhava) [inline bin/build-kubernetes-images.sh:14]: We already have SPARK_VERSION exposed in https://github.com/h2oai/sparkling-water/blob/b67d34a2ba5391cf7eb25397fb84c15f2239a809/bin/sparkling-env.sh#L146 but uses value from gradle.properties.

I guess the problem was with minor version?

We have this line to export the installed SPARK_VERSION https://github.com/h2oai/sparkling-water/blob/b67d34a2ba5391cf7eb25397fb84c15f2239a809/bin/sparkling-env.sh#L126

Maybe we can expose it in sparkling-env as `INSTALLED_SPARK_VERSION` ? WDYT?

File: bin/build-kubernetes-images.sh [hunk 2]
```diff
@@ -24,27 +25,27 @@ if [ "$1" = "scala" ]; then
...
-  docker build -t "sparkling-water-r:$VERSION" -f "$WORKDIR/Dockerfile-R" "$WORKDIR"
+  docker build --build-arg spark_version=$SPARK_VERSION -t "sparkling-water-r:$VERSION" -f "$WORKDIR/Dockerfile-R" "$WORKDIR"
   echo "Done!"
 fi
 
 echo "Cleaning up temporary directories"
 rm -rf "$WORKDIR"
 
-echo "All done! You can find your images by running: docker images"
+echo "All done! You can find your images by running: docker images"
```
Comment (jakubhava) [inline bin/build-kubernetes-images.sh:51]: nit: missing nl

File: kubernetes/build.gradle [hunk 3]
```diff
@@ -38,9 +40,13 @@ task createRDockerfile(type: Dockerfile) {
 
 task createPythonDockerfile(type: Dockerfile) {
   destFile = outputFilePython
-  from "spark-py:${sparkVersion}"
+  arg "spark_version"
+  from 'spark-py:$spark_version'
+  arg "spark_version"
   copyFile("h2o_pysparkling_${sparkMajorVersion}-${version}.zip", "/opt/spark/pyspark/python/lib/h2o_pysparkling_${sparkMajorVersion}-${version}.zip")
-  runCommand "pip install /opt/spark/pyspark/python/lib/h2o_pysparkling_${sparkMajorVersion}-${version}.zip"
+  runCommand 'echo "pyspark==$spark_version" > constraints'
```
Comment (jakubhava) [inline kubernetes/build.gradle:47]: We should run all the last 3 `runCommand` inside one `runCommand`. This way docker greates 3 layers, one with contastrains file, one with the packages and one with the removed constrains file -> bigger size.

If we do all the changes in one scope, docker won't save the intermediate files

Comment (mn-mikke) [reply kubernetes/build.gradle:47]: ok, thanks!

File: bin/build-kubernetes-images.sh (hunk 1 above)
Comment (mn-mikke) [reply bin/build-kubernetes-images.sh:14]: ok, `installed_spark_version` sounds good to me. Will try to use.

yes, the problem i see is the minor version. IMHO, a user should be able create image of any spark version.

Comment (jakubhava) [reply bin/build-kubernetes-images.sh:14]: Yup, absolutely agree

File: kubernetes/build.gradle [hunk 4]
```diff
@@ -38,9 +40,13 @@ task createRDockerfile(type: Dockerfile) {
...
+  runCommand "pip install /opt/spark/pyspark/python/lib/h2o_pysparkling_${sparkMajorVersion}-${version}.zip -c constraints"
```
Comment (mn-mikke) [inline kubernetes/build.gradle:48]: this line instlalls all depencies including pyspark. Shouldn't pyspark be omitted if it's already in the image?

Comment (jakubhava) [reply kubernetes/build.gradle:48]: Yup, great catch! 

I would just verify after this change that the image works as expected -> starting pysparkling works

File: bin/build-kubernetes-images.sh (hunk 1 above)
Comment (mn-mikke) [reply bin/build-kubernetes-images.sh:14]: The installed version doesn't contain spark fix version part, just `major.minor`. Should we expect that spark-submit is on class_path or take it from `$SPARK_HOME`

Comment (jakubhava) [reply bin/build-kubernetes-images.sh:14]: The sparkling-env script has method `checkSparkHome` which finds SPARK_HOME, we should follow that logic.

We can create new properties such as INSTALLED_SPARK_MAJOR_MINOR and full version as well

Comment (mn-mikke) [reply bin/build-kubernetes-images.sh]: ok

**Pull request merged**